### PR TITLE
Fix `auto-id` build for backwards compatibility with React <18

### LINKED ISSUE #921: 0.17.0 does not work with React 17
## 🐛 Bug report

Since useId is not available on React17 it breaks with reach@0.17.0

In auto-id package https://github.com/reach/reach-ui/blob/develop/packages/auto-id/src/index.tsx,  

```import * as React from "react";``` 

is compiled as 

```import { useId as useId$1, useState, useEffect } from 'react'```

 in the esm module.

This leads to the following error 

image

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Reach Package    |  reach/alert-dialog   |    0.17.0     |
| React            |     React    |   17.0.2      |
| React-scripts           |     react-scripts    |  5.0.0     |
| npm/yarn         |     yarn     |   1.22.18      |
| Operating System |    MacOs     |     12.3.1    |

### PR DESCRIPTION
**Pull request by emmenko**
## Summary

Fixes #921

TL;DR: the bundle of `@reach/auto-id` contains an import of `useId` from `react`. For versions of React <18 this import does not exist, causing an error when this package is used.

image

This can also be seen in the Storybook playground build output (although Storybook shows a warning)

image

To fix that, I suggest to remove that code and change the implementation once React >=18 is supported.

I also fixed another import issue shown by Storybook

image

---

Thank you for contributing to Reach UI! Please fill in this template before submitting your PR to help us process your request more quickly.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code (Compile and run).
- [ ] Add or edit tests to reflect the change (Run with `yarn test`).
- [x] Add or edit Storybook examples to reflect the change (Run with `yarn start`).
- [x] Ensure formatting is consistent with the project's Prettier configuration.
- [ ] Add documentation to support any new features.

This pull request:

- [ ] Creates a new package
- [x] Fixes a bug in an existing package
- [ ] Adds additional features/functionality to an existing package
- [x] Updates documentation or example code
- [ ] Other

If creating a new package:

- [ ] Make sure the new package directory contains each of the following, and that their structure/formatting mirrors other related examples in the project:
  - [ ] `examples` directory
  - [ ] `src` directory with an `index.tsx` entry file
  - [ ] At least one example file per feature introduced by the new package
  - [ ] Base styles in a `style.css` file (if needed by the new package)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,9 @@ node_modules
 .env
 coverage
 
+# storybook
+/playground/storybook-static
```
Comment (emmenko) [inline .gitignore:16]: Artifact of building Storybook locally.

File: packages/auto-id/src/auto-id.ts [hunk 2]
```diff
@@ -87,15 +87,6 @@ function useId(
...
 
 function useId(providedId?: number | string | undefined | null) {
-	// TODO: Remove error flag when updating internal deps to React 18. None of
-	// our tricks will play well with concurrent rendering anyway.
-	// @ts-expect-error
-	if (typeof React.useId === "function") {
-		// @ts-expect-error
-		let id = React.useId(providedId);
-		return providedId != null ? providedId : id;
-	}
```
Comment (emmenko) [inline packages/auto-id/src/auto-id.ts:97]: This is what is causing the problem. I suggest to simply remove this once only React >=18 is supported.

File: playground/stories/window-size/basic.example.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import * as React from "react";
-import WindowSize from "@reach/window-size";
+import { WindowSize } from "@reach/window-size";
```
Comment (emmenko) [inline playground/stories/window-size/basic.example.js:2]: The import was wrong

File: packages/auto-id/src/auto-id.ts (hunk 2 above)
Comment (zkuzmic) [reply packages/auto-id/src/auto-id.ts:97]: Is there no way to rewrite this to keep your build process from showing a warning? I think that would be preferable.
